project/categories/serializers.py

```diff
@@ -0,0 +1,33 @@
+from rest_framework import serializers
+
+
+class CategoryListSerializer(serializers.ModelSerializer):
+    """ """
+    class Meta:
+        model = Category
+        fields = ("id", "name")
+
+
+class CategorySerializer(serializers.ModelSerializer):
+    """ """
+    preferred = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Category
+        fields = ("id", "name", "preferred")
+
+    def get_preferred(self, obj):
+        user = None
+        request = self.context.get("request")
+
+        # Check for authenticated user
+        if request and hasattr(request, "user"):
+            user = request.user
+        else:
+            return True
+
+        if user.is_anonymous():
+            return True
+
+        account = Profile.objects.get(user=user)
+        return obj.id in account.categories.values_list("id", flat=True)
```

Comment:
@anurag629 I believe this will give an error as the Profile model has not been imported.

Reply:
@anurag629 please add the necessary imports for this code to work. I have suggested a change to the top of the file with the required imports.